File: JS/383-ransom-note.js
// hist
/**
 * @param {string} ransomNote
 * @param {string} magazine
 * @return {boolean}
 */
var canConstruct = function(ransomNote, magazine) {
    let hist = Array(26).fill(0);
    for (let c of magazine) hist[c.codePointAt() - 'a'.codePointAt()]++;
    for (let c of ransomNote) hist[c.codePointAt() - 'a'.codePointAt()]--;
    if (hist.findIndex(cnt => cnt < 0) == -1) return true;
    else return false;
};

File: C++/383-ransom-note.cpp
#include <string>

using namespace std;

class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
        int chars[26] = {};
        for (char c : magazine) {
            chars[c - 'a']++;
        }
        for (char c : ransomNote) {
            chars[c - 'a']--;
            if (chars[c - 'a'] < 0) return false;
        }
        return true;
    }
};